lib/quicly.c

```diff
@@ -1417,9 +1417,20 @@ static int received_key_update(quicly_conn_t *conn, uint64_t newly_decrypted_key
     }
 }
 
+static void dump_conn_stats(quicly_conn_t *conn) {
+#if QUICLY_USE_EMBEDDED_PROBES || QUICLY_USE_DTRACE
+    if (!QUICLY_CONN_STATS_ENABLED())
+        return;
+    quicly_stats_t stats;
+    quicly_get_stats(conn, &stats);
+    QUICLY_PROBE(CONN_STATS, conn, conn->stash.now, &stats, sizeof(stats));
+#endif
+}
+
 void quicly_free(quicly_conn_t *conn)
 {
     QUICLY_PROBE(FREE, conn, conn->stash.now);
+    dump_conn_stats(conn);
```

Comment:
Instead of creating a separate function, how about doing something like below here?

The rationale is that we do not want to dump the connection stats at an arbitrary moment in the current form, because to use the stats in an meaningful way, we need to understand when they are collected. To paraphrase, they are tied to the `quicly_free` function.

```c
#if QUICLY_USE_EMBEDDED_PROBES ||QUICLY_USE_DTRACE
    if (QUICLY_CONN_STATS_ENABLED()) {
        quicly_stats_t stats;
        quicly_get_stats(conn, &stats);
        QUICLY_PROBE(...);
    }
#endif
```

Reply:
Hm. That's true for now, but we may want to grab the stats at other times, say on idle periods. But, let's deal with that when we get to it. I'll remove the function for now.